src/mbgl/style/conversion/filter.cpp

```diff
@@ -53,9 +53,14 @@ bool isExpression(const Convertible& filter) {
         optional<std::string> operand = toString(arrayMember(filter, 1));
         return operand && *operand != "$id" && *operand != "$type";
 
-    } else if (*op == "in" || *op == "!in" || *op == "!has" || *op == "none") {
+    } else if (*op == "!in" || *op == "!has" || *op == "none") {
         return false;
 
+    } else if (*op == "in") {
+        optional<std::string> str = toString(arrayMember(filter, 1));
+
+        return arrayLength(filter) >= 3 && (!str || isArray(arrayMember(filter, 2)));
```

Comment:
While you’re at it, can you incorporate the change in mapbox/mapbox-gl-js#9374 too?

Reply:
I did incorporate that change.

Reply:
Ah, my bad, I overlooked that part because of the variable name `str`, which I misunderstood at first.